modules/images/webp-uploads/load.php

```diff
@@ -162,7 +168,10 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 				continue;
 			}
 
-			if ( webp_uploads_should_discard_additional_image_file( $properties, $source ) ) {
+			if (
+				! empty( $metadata['sources'][ $mime_type ] ) &&
+				webp_uploads_should_discard_additional_image_file( $properties['sources'][ $mime_type ], $source )
+			) {
```

Comment:
While I agree on the similar change further above, I think here it shouldn't be needed: `$properties` for every image size should already have a `filesize` key from WP core, and that image should be for the primary (typically original) MIME type. Again, similar to almost all other problems here, I think the issue here may be what I outlined in my very long comment above.

Reply:
I don't know why but it wasn't the case when I tested it yesterday. Here is a var_dump result for metadata of an image in one of the tests:

```php
array(8) {
  ["width"]=>
  int(1080)
  ["height"]=>
  int(720)
  ["file"]=>
  string(21) "2022/09/leafs-572.jpg"
  ["sizes"]=> ...
  ["image_meta"]=> ...
  ["dominant_color"]=>
  string(6) "283c23"
  ["has_transparency"]=>
  bool(false)
  ["sources"]=> ...
}
```

As you can see there is no `filesize` property on the root level, that's why I updated the functionality to use the correct path to main source information that contains `filesize` data. If the `filesize` key is generated by WP core, then I wonder why it doesn't do it in phpunit tests 🤔

Reply:
@eugene-manuilov This is indeed odd, not sure why it is not included. See my comment https://github.com/WordPress/performance/pull/527#discussion_r971131216, we need to look into this further.

On another note, you're referring to the overall metadata above, but this particular comment here is about each individual size's data, not the overall metadata. The sizes should each also have a `filesize` key, as long as they were uploaded with WordPress 6.0. (Of course we have to handle this gracefully, for images uploaded in an older WordPress version.)

Reply:
Yes, we don't need this. Reverted.